src/InContextSignup.js

```diff
@@ -32,23 +34,50 @@ export class InContextSignup {
         return isValidTLD() && !isLocalNetwork()
     }
 
-    isAvailable () {
-        const isEnabled = this.device.settings?.featureToggles.emailProtection_incontext_signup
-        const isLoggedIn = this.device.isDeviceSignedIn()
-        return isEnabled && !isLoggedIn && !this.isPermanentlyDismissed() && this.isOnValidDomain() && this.isInstalledRecently
+    isOnAssetsPage () {
+        // Check if on locally hosted macOS autofill page
+        return window.location.href.endsWith('assets/TopAutofill.html')
+    }
+
+    isAllowedByDevice () {
+        if (typeof this.isInstalledRecently === 'boolean') {
+            return this.isInstalledRecently
+        } else {
+            // Don't restrict in-context signup based on recent installation
+            // if the device hasn't provided a clear indication
+            return true
+        }
+    }
+
+    /**
+     * @param {import('./Form/matching.js').SupportedTypes} [inputType]
+     * @returns {boolean}
+     */
+    isAvailable (inputType = 'identities.emailAddress') {
```

Comment:
From a cursory look it seems that if we use `emailAddress` (without the `identities.` prefix) we could simplify some of the data passing in the callers (no need to import new dependencies on those files and sometimes the piece of data may already be available). Any reason not to?

Reply:
Good question, I think I went down this route because `src/UI/controllers/HTMLTooltipUIController.js` only has the full input type (but of course we can derive the subtype). Not sure it simplifies much (there's still places we need to import), but it keeps us aligned with the current condition `subtype === 'emailAddress' && ...` in `src/Form/inputTypeConfig.js` -- added a commit to change this 👍